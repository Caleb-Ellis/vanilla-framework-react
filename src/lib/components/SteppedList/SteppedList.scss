@import 'vanilla-framework/scss/base';
@import 'vanilla-framework/scss/patterns_lists';
@import 'vanilla-framework/scss/patterns_grid';
@import 'vanilla-framework/scss/utilities_clearfix';
@include vf-b-typography;
@include vf-b-lists;
@include vf-p-lists;
@include vf-p-grid;
@include vf-u-clearfix;
