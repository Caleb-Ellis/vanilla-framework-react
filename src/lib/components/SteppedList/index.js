export { default as SteppedList } from './SteppedList';
export { default as SteppedListItem } from './SteppedListItem';
