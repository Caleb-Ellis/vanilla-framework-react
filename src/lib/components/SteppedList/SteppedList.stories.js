import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import SteppedList from './SteppedList';
import SteppedListItem from './SteppedListItem';

storiesOf('Stepped List', module)
  .add('Default',
    withInfo('If you want to display a list of items that form a set of steps — like a tutorial or instructions — you can use the SteppedList component. This component is best used in a <Strip light> component as the description sections are displayed in a white box.')(() => (
      <div style={{ maxWidth: '45em' }}>
        <SteppedList>
          <SteppedListItem
            title="First step"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed purus lorem,
            dictum vel dolor eu, tincidunt suscipit magna. Suspendisse dignissim nisl
            vitae turpis iaculis, ut tempor enim gravida.
          </SteppedListItem>
          <SteppedListItem
            title="Second step"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed purus lorem,
            dictum vel dolor eu, tincidunt suscipit magna. Suspendisse dignissim nisl
            vitae turpis iaculis, ut tempor enim gravida.
          </SteppedListItem>
          <SteppedListItem
            title="Third step"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed purus lorem,
            dictum vel dolor eu, tincidunt suscipit magna. Suspendisse dignissim nisl
            vitae turpis iaculis, ut tempor enim gravida.
          </SteppedListItem>
          <SteppedListItem
            title="Last but not least"
            bullet="99"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed purus lorem,
            dictum vel dolor eu, tincidunt suscipit magna. Suspendisse dignissim nisl
            vitae turpis iaculis, ut tempor enim gravida.
          </SteppedListItem>
        </SteppedList>
      </div>),
    ),
  )
  .add('Detailed',
    withInfo('Add the detailed modifier to separate content on either side of the container.')(() => (
      <div style={{ maxWidth: '45em' }}>
        <SteppedList detailed>
          <SteppedListItem
            title="First step"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed purus lorem,
            dictum vel dolor eu, tincidunt suscipit magna. Suspendisse dignissim nisl
            vitae turpis iaculis, ut tempor enim gravida.
          </SteppedListItem>
          <SteppedListItem
            title="Second step"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed purus lorem,
            dictum vel dolor eu, tincidunt suscipit magna. Suspendisse dignissim nisl
            vitae turpis iaculis, ut tempor enim gravida.
          </SteppedListItem>
          <SteppedListItem
            title="Third step"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed purus lorem,
            dictum vel dolor eu, tincidunt suscipit magna. Suspendisse dignissim nisl
            vitae turpis iaculis, ut tempor enim gravida.
          </SteppedListItem>
          <SteppedListItem
            title="Last but not least"
            bullet="99"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed purus lorem,
            dictum vel dolor eu, tincidunt suscipit magna. Suspendisse dignissim nisl
            vitae turpis iaculis, ut tempor enim gravida.
          </SteppedListItem>
        </SteppedList>
      </div>),
    ),
  );
