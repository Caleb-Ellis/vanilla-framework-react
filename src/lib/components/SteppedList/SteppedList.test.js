import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import SteppedList from './SteppedList';
import SteppedListItem from './SteppedListItem';

describe('SteppedList component', () => {
  it('should render a simple stepped list correctly', () => {
    const steppedList = ReactTestRenderer.create(
      <SteppedList>
        <SteppedListItem
          title="First step"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed purus lorem,
          dictum vel dolor eu, tincidunt suscipit magna. Suspendisse dignissim nisl
          vitae turpis iaculis, ut tempor enim gravida.
        </SteppedListItem>
        <SteppedListItem
          title="Second step"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed purus lorem,
          dictum vel dolor eu, tincidunt suscipit magna. Suspendisse dignissim nisl
          vitae turpis iaculis, ut tempor enim gravida.
        </SteppedListItem>
        <SteppedListItem
          title="Third step"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed purus lorem,
          dictum vel dolor eu, tincidunt suscipit magna. Suspendisse dignissim nisl
          vitae turpis iaculis, ut tempor enim gravida.
        </SteppedListItem>
        <SteppedListItem
          title="Last but not least"
          bullet="99"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed purus lorem,
          dictum vel dolor eu, tincidunt suscipit magna. Suspendisse dignissim nisl
          vitae turpis iaculis, ut tempor enim gravida.
        </SteppedListItem>
      </SteppedList>,
    );
    const json = steppedList.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render a detailed stepped list correctly', () => {
    const steppedList = ReactTestRenderer.create(
      <SteppedList detailed>
        <SteppedListItem
          title="First step"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed purus lorem,
          dictum vel dolor eu, tincidunt suscipit magna. Suspendisse dignissim nisl
          vitae turpis iaculis, ut tempor enim gravida.
        </SteppedListItem>
        <SteppedListItem
          title="Second step"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed purus lorem,
          dictum vel dolor eu, tincidunt suscipit magna. Suspendisse dignissim nisl
          vitae turpis iaculis, ut tempor enim gravida.
        </SteppedListItem>
        <SteppedListItem
          title="Third step"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed purus lorem,
          dictum vel dolor eu, tincidunt suscipit magna. Suspendisse dignissim nisl
          vitae turpis iaculis, ut tempor enim gravida.
        </SteppedListItem>
        <SteppedListItem
          title="Last but not least"
          bullet="99"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed purus lorem,
          dictum vel dolor eu, tincidunt suscipit magna. Suspendisse dignissim nisl
          vitae turpis iaculis, ut tempor enim gravida.
        </SteppedListItem>
      </SteppedList>,
    );
    const json = steppedList.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should accept generic list items', () => {
    const steppedList = ReactTestRenderer.create(
      <SteppedList>
        <li>One</li>
        <li>Two</li>
        <li>Three</li>
      </SteppedList>,
    );
    const json = steppedList.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should not render anything if no children present', () => {
    const steppedList = ReactTestRenderer.create(
      <SteppedList />,
    );
    const json = steppedList.toJSON();
    expect(json).toMatchSnapshot();
  });
});
