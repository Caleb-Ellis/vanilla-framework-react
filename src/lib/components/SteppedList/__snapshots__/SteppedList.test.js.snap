// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SteppedList component should accept generic list items 1`] = `
<ol
  className="p-list-step"
>
  <li
    detailed={false}
    index={0}
  >
    One
  </li>
  <li
    detailed={false}
    index={1}
  >
    Two
  </li>
  <li
    detailed={false}
    index={2}
  >
    Three
  </li>
</ol>
`;

exports[`SteppedList component should not render anything if no children present 1`] = `null`;

exports[`SteppedList component should render a detailed stepped list correctly 1`] = `
<ol
  className="p-stepped-list--detailed"
>
  <li
    className="p-list-step__item"
  >
    <h3
      className="p-list-step__title"
    >
      <span
        className="p-list-step__bullet"
      >
        1
      </span>
      First step
    </h3>
    <p
      className="p-list-step__content"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed purus lorem, dictum vel dolor eu, tincidunt suscipit magna. Suspendisse dignissim nisl vitae turpis iaculis, ut tempor enim gravida.
    </p>
  </li>
  <li
    className="p-list-step__item"
  >
    <h3
      className="p-list-step__title"
    >
      <span
        className="p-list-step__bullet"
      >
        2
      </span>
      Second step
    </h3>
    <p
      className="p-list-step__content"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed purus lorem, dictum vel dolor eu, tincidunt suscipit magna. Suspendisse dignissim nisl vitae turpis iaculis, ut tempor enim gravida.
    </p>
  </li>
  <li
    className="p-list-step__item"
  >
    <h3
      className="p-list-step__title"
    >
      <span
        className="p-list-step__bullet"
      >
        3
      </span>
      Third step
    </h3>
    <p
      className="p-list-step__content"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed purus lorem, dictum vel dolor eu, tincidunt suscipit magna. Suspendisse dignissim nisl vitae turpis iaculis, ut tempor enim gravida.
    </p>
  </li>
  <li
    className="p-list-step__item"
  >
    <h3
      className="p-list-step__title"
    >
      <span
        className="p-list-step__bullet"
      >
        99
      </span>
      Last but not least
    </h3>
    <p
      className="p-list-step__content"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed purus lorem, dictum vel dolor eu, tincidunt suscipit magna. Suspendisse dignissim nisl vitae turpis iaculis, ut tempor enim gravida.
    </p>
  </li>
</ol>
`;

exports[`SteppedList component should render a simple stepped list correctly 1`] = `
<ol
  className="p-list-step"
>
  <li
    className="p-list-step__item"
  >
    <h3
      className="p-list-step__title"
    >
      <span
        className="p-list-step__bullet"
      >
        1
      </span>
      First step
    </h3>
    <p
      className={null}
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed purus lorem, dictum vel dolor eu, tincidunt suscipit magna. Suspendisse dignissim nisl vitae turpis iaculis, ut tempor enim gravida.
    </p>
  </li>
  <li
    className="p-list-step__item"
  >
    <h3
      className="p-list-step__title"
    >
      <span
        className="p-list-step__bullet"
      >
        2
      </span>
      Second step
    </h3>
    <p
      className={null}
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed purus lorem, dictum vel dolor eu, tincidunt suscipit magna. Suspendisse dignissim nisl vitae turpis iaculis, ut tempor enim gravida.
    </p>
  </li>
  <li
    className="p-list-step__item"
  >
    <h3
      className="p-list-step__title"
    >
      <span
        className="p-list-step__bullet"
      >
        3
      </span>
      Third step
    </h3>
    <p
      className={null}
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed purus lorem, dictum vel dolor eu, tincidunt suscipit magna. Suspendisse dignissim nisl vitae turpis iaculis, ut tempor enim gravida.
    </p>
  </li>
  <li
    className="p-list-step__item"
  >
    <h3
      className="p-list-step__title"
    >
      <span
        className="p-list-step__bullet"
      >
        99
      </span>
      Last but not least
    </h3>
    <p
      className={null}
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed purus lorem, dictum vel dolor eu, tincidunt suscipit magna. Suspendisse dignissim nisl vitae turpis iaculis, ut tempor enim gravida.
    </p>
  </li>
</ol>
`;
